### PR TITLE
Misc nvse requests

### PR DESCRIPTION
**Pull request by bdemmy**
Think these hooks are sufficient.  Lmk if they are not.

### CODE REVIEW

**Comment by korri123** [comment]
The hook structure looks quite unclean, if possible we prefer avoiding inline assembly with the help of GetParentBasePtr and prefer hooking calls directly (if thiscall is used __fastcall can be used to emulate it). If inline assembly is required we generally try to avoid pushad and popad. We also generally try not to mix inline assembly with C++.
Example: https://github.com/xNVSE/NVSE/blob/a2852dd1f8a686d7506c44632559cc7b5d9726c5/nvse/nvse/Hooks_SaveLoad.cpp#L70
(feel free to checkout other usages of GetParentBasePtr, some good ones are in xNVSE, kNVSE and ShowOff xNVSE)

On another note, this seems to register events for NVSE plugins but they won't be available as event handlers for scripts. Not sure how useful this is for plugins as an event handler. If it was intended for scripts there's a framework Demorome and me designed within xNVSE for registering new events for scripts, Demorome knows it a bit better though.

**Comment by korri123** [comment]
Actually, the only message dispatched is to the plugin manager so it's probably missing the call to `EventManager::DispatchEvent` as well if it should work in script events

**Comment by bdemmy** [comment]
This is all cleaned up now. Just going to convert to the other event system. Thanks again :)

**Comment by bdemmy** [comment]
Added some lockpick hooks (Lock broken, success, pick break).

**Comment by bdemmy** [comment]
Added terminal hack fail/success events. More to come.

File: nvse/nvse/Hooks_Other.cpp [hunk 1]
```diff
@@ -179,6 +180,84 @@ namespace OtherHooks
...
+		}
+
+		void __fastcall OnLockPickSuccess(TESObjectREFR* doorRef) {
+			EventManager::DispatchEvent("onlockpicksuccess", nullptr, doorRef);
+
+			ThisStdCall(0x5692A0, doorRef);
+		}
+
+		void __fastcall OnLockPickBroken() {
+			auto* ebp = GetParentBasePtr(_AddressOfReturnAddress(), false);
```
Comment (Demorome) [inline nvse/nvse/Hooks_Other.cpp:222]: It would probably be cleaner to get the LockPickMenu from the `g_lockPickMenu` global at `0x11DA204`.

Comment (bdemmy) [reply nvse/nvse/Hooks_Other.cpp]: Done, thanks

**Pull request merged**